Apps/Src/ReadCarCAN.c

```diff
@@ -40,23 +35,22 @@ static uint8_t oldestMsgIdx = 0;
 static uint8_t SOC = 0;
 static uint32_t SBPV = 0;
 
-
-// Handler to turn array back on
-static void arrayRestart(void *p_tmr, void *p_arg); 
-
-
 // Getter function for chargeEnable
-bool ChargeEnable_Get() 
-{
+bool ChargeEnable_Get(){
     return chargeEnable;
 }
 
-
 /**
  * @brief adds new messages by overwriting old messages in the saturation buffer and then updates saturation
  * @param chargeMessage whether bps message was charge enable (1) or disable (-1)
 */
 static void updateSaturation(int8_t chargeMessage){
+    if(chargeMessage == 1){
+        chargeEnable = true;
+    }else{
+        chargeEnable = false;
+    }
+
```

Comment:
I don't think this has the intended effect. This will turn on chargeEnable if we get a chargeMessage of 1 and turn it off if we get a chargeMessage of 0. Ideally, we want chargeEnable to be turned on only if the saturation buffer is above the saturation threshold. If we get a chargeMessage of 0, we want to decrement the buffer and immediately turn chargeEnable off. If we get a chargeMessage of 1, we want to increment the buffer and only turn on chargeEnable if we're past the threshold.